packages/core/admin/admin/src/pages/SettingsPage/pages/ApiTokens/EditView/index.js

```diff
@@ -238,6 +244,66 @@ const ApiTokenCreateView = () => {
+                        <GridItem key="duration" col={6} xs={12}>
+                          <Select
+                            name="duration"
+                            label={formatMessage({
+                              id: 'Settings.apiTokens.form.duration',
+                              defaultMessage: 'Token duration',
+                            })}
+                            value={values.duration}
+                            error={
+                              errors.duration
+                                ? formatMessage(
+                                    errors.duration?.id
+                                      ? errors.duration
+                                      : { id: errors.duration, defaultMessage: errors.duration }
+                                  )
+                                : null
+                            }
+                            onChange={value => {
+                              handleChange({ target: { name: 'duration', value } });
+                            }}
+                            required
+                            disabled={!isCreating}
+                          >
+                            <Option value="7">
+                              {formatMessage({
+                                id: 'Settings.apiTokens.duration.7-days',
+                                defaultMessage: '7 days',
+                              })}
+                            </Option>
+                            <Option value="30">
+                              {formatMessage({
+                                id: 'Settings.apiTokens.duration.30-days',
+                                defaultMessage: '30 days',
+                              })}
+                            </Option>
+                            <Option value="90">
+                              {formatMessage({
+                                id: 'Settings.apiTokens.duration.90-days',
+                                defaultMessage: '90 days',
+                              })}
+                            </Option>
+                            <Option value="unlimited">
+                              {formatMessage({
+                                id: 'Settings.apiTokens.duration.unlimited',
+                                defaultMessage: 'Unlimited',
+                              })}
+                            </Option>
+                          </Select>
+                          <Typography variant="pi" textColor="neutral600">
+                            {!isCreating &&
+                              `${formatMessage({
+                                id: 'Settings.apiTokens.duration.expiration-date',
+                                defaultMessage: 'Expiration date',
+                              })}: ${getDateOfExpiration(
+                                apiToken?.createdAt,
+                                values.duration,
+                                localStorage.getItem('strapi-admin-language')
```

Comment:
@ronronscelestes asked me about an opinion here: reading the entry from localStorage on each render seems very expensive to me. What do you think about reading the language onceoutside the component and then re-using the cached value?

The `AssetPreview` contains an example how this is done in other places of the code, but I'm not sure I am a fan of it:

https://github.com/strapi/strapi/blob/e50c88e766f4547f57556a68dc71c9beeae8e2af/packages/core/upload/admin/src/components/EditAssetDialog/PreviewBox/AssetPreview.js#L17

Reply:
I reused the same logic for this in AssetPreview @gu-stav